test/integration/app/libraries/iis_app.rb

```diff
@@ -0,0 +1,124 @@
+# encoding: utf-8
+# frozen_string_literal: true
+# check for site in IIS
+# Usage:
+# describe iis_app('/myapp', 'Website') do
+#   it { should exist }
+#   it { should have_application_pool('MyAppPool') }
+#   it { should have_protocols('http') }
+#   it { should have_site_name('Default Web Site') }
+#   it { should have_physical_path('C:\\inetpub\\wwwroot\\myapp') }
+#   it { should have_path('\\My Application') }
+# end
+#
+# Note: this is only supported in windows 2012 and later
+
+class IisApp < Inspec.resource(1)
+  name 'iis_app'
+  desc 'Tests IIS application configuration on windows. Supported in server 2012+ only'
+  example "
+    describe iis_app('/myapp', 'Default Web Site') do
+      it { should exist }
+      it { should have_application_pool('MyAppPool') }
+      it { should have_protocols('http') }
+      it { should have_site_name('Default Web Site') }
+      it { should have_physical_path('C:\\inetpub\\wwwroot\\myapp') }
+      it { should have_path('\\My Application') }
+    end
+  "
+
+  def initialize(path, site_name)
+    @path = path
+    @site_name = site_name
+    @cache = nil
+
+    @app_provider = AppProvider.new(inspec)
+
+    # verify that this resource is only supported on Windows
+    return skip_resource 'The `iis_app` resource is not supported on your OS.' if inspec.os[:family] != 'windows'
+  end
+
+  def application_pool
+    iis_app[:application_pool]
```

Comment:
if `iis_app` is `nil` because AppProvider is unable to parse the JSON output, I believe these methods will return `undefined method '[]' for nil:NilClass` which is pretty annoying for end-users as I'm sure you'll agree. 🙂 

Would it be better to change iis_app to default return an empty hash `{}`? That will avoid the unnecessary exceptions, and your `has_something?` methods can remove the ternary checks (since if it's an empty hash, `nil` won't equal the condition you're checking.